File: ML A-Z/Part 1 - Data Preprocessing/dataprep.r
setwd('C:\\Users\\chaos\\Documents\\ML A-Z\\Part 1 - Data Preprocessing')

#Importing the dataset
data = read.csv('Data.csv')

#Taking care of missing data
data$Age = ifelse(is.na(data$Age),
                  ave(data$Age, FUN = function(x) mean(x, na.rm = TRUE)),
                  data$Age)

data$Salary = ifelse(is.na(data$Salary),
                  ave(data$Salary, FUN = function(x) mean(x, na.rm = TRUE)),
                  data$Salary)

# Encoding categorical data
data$Country = factor(data$Country,
                        levels = c('France', 'Spain', 'Germany'), 
                        labels = c(1,2,3))

data$Purchased = factor(data$Purchased,
                      levels = c('No', 'Yes'),
                      labels = c(0,1))

# Splitting the dataset into the training set and test set
#install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(data$Purchased, SplitRatio = 0.8)
training_set = subset(data, split == TRUE)
test_set = subset(data, split == FALSE)

#feature scaling
training_set[,2:3] = scale(training_set[,2:3])
test_set[,2:3] = scale(test_set[,2:3])

File: ML A-Z/Part 1 - Data Preprocessing/dataprep.py
# -*- coding: utf-8 -*-
"""
Spyder Editor

This is a temporary script file.
"""

#Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Importing the dataset
data = pd.read_csv('Data.csv')
X = data.iloc[:, :-1].values
y = data.iloc[:, -1:].values

#taking care of missing data
from sklearn.impute import SimpleImputer
imp = SimpleImputer()
imp = imp.fit(X[:, 1:3])
X[:, 1:3] = imp.transform(X[:, 1:3])

#encoding categorical data
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
labelencoder_X = LabelEncoder()
X[:, 0] = labelencoder_X.fit_transform(X[:, 0])

onehotencoder = OneHotEncoder(categorical_features=[0])
X = onehotencoder.fit_transform(X).toarray()

labelencoder_y = LabelEncoder()
y = labelencoder_y.fit_transform(y)

#splitting the dataset into the training set and test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state = 0)

#feature scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)

pd.DataFrame(X).head()
pd.DataFrame(y).head()